includes/class-wc-gateway-swedbank-pay-checkout.php

```diff
@@ -0,0 +1,1837 @@
+				'shipping_phone'        => $customer->get_billing_phone(),
+			] );
+
+			if ( is_wp_error( $order_id ) ) {
+				throw new Exception( $order_id->get_error_message() );
+			}
+
+			// Store Order ID in session so it can be re-used after payment failure
+			WC()->session->set( 'order_awaiting_payment', $order_id );
+		}
+
+		// Get Customer Profile
+		$profile = $this->get_consumer_profile( get_current_user_id() );
+
+		// Initiate Payment Order
+		$_POST['swedbank_pay_customer_reference'] = $profile['reference'];
+		$result = $this->process_payment( $order_id );
+		if ( is_array( $result ) && isset( $result['js_url'] ) ) {
+			WC()->session->set( 'swedbank_pay_checkout_js_url', $result['js_url'] );
+		} else {
+			WC()->session->__unset( 'swedbank_pay_checkout_js_url' );
+		}
+	}
+
+	/**
+	 * Hook before_checkout_billing_form
+	 *
+	 * @param $checkout
+	 */
+	public function before_checkout_billing_form( $checkout ) {
+		if ( $this->instant_checkout !== 'yes' ) {
+			return;
+		}
+
+		if ( $this->checkin !== 'yes' ) {
+			return;
+		}
+
+		// Get saved consumerProfileRef
+		$profile = $this->get_consumer_profile( get_current_user_id() );
+
+		// Initiate consumer session to obtain consumerProfileRef after checkin
+		$js_url = $profile['url'];
+		if ( empty( $profile['reference'] ) ) {
+			// Initiate consumer session
+			$params = [
+				'operation'           => 'initiate-consumer-session',
+				'consumerCountryCode' => $this->checkin_country,
+			];
+
+			try {
+				$result = $this->request( 'POST', '/psp/consumers', $params );
+				$js_url = self::get_operation( $result['operations'], 'view-consumer-identification' );
+			} catch ( Exception $e ) {
+				$profile['reference'] = null;
+				$profile['billing'] = null;
+			}
+		}
+
+		WC()->session->set( 'consumer_js_url', $js_url );
+
+		// Checkin Form
+		wc_get_template(
+			'checkout/swedbank-pay/checkin.php',
+			[
+				'checkin_country'  => $this->checkin_country,
+				'selected_country' => apply_filters( 'swedbank_pay_checkin_default_country', 'SE' ),
+				'js_url'           => $js_url,
+				'consumer_data'    => $profile['billing'],
+				'consumer_profile' => $profile['reference']
+			],
+			'',
+			dirname( __FILE__ ) . '/../templates/'
+		);
+	}
+
+	/**
+	 * Render Payment Methods HTML.
+	 *
+	 * @return void
+	 */
+	public function woocommerce_checkout_payment() {
+		if ( $this->enabled === 'yes' && $this->instant_checkout === 'yes' ) {
+			$js_url = WC()->session->get( 'swedbank_pay_checkout_js_url' );
+
+			wc_get_template(
+				'checkout/swedbank-pay/payment.php',
+				[
+					//'checkout' => WC()->checkout()
+					'js_url' => $js_url
+				],
+				'',
+				dirname( __FILE__ ) . '/../templates/'
+			);
+		}
+	}
+
+	/**
+	 * Ajax: Retrieve Address
+	 *
+	 * @return void
+	 */
+	public function ajax_swedbank_pay_checkout_get_address() {
+		check_ajax_referer( 'swedbank_pay_checkout', 'nonce' );
+
+		$type = isset( $_POST['type'] ) ? wc_clean( $_POST['type'] ) : '';
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Direct use of $_POST Superglobal detected.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-payments/pullRequest?prid=5340032)